src/app/options/qgsideoptions.cpp

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+    qgsideoptions.cpp
+    -------------------------
+    begin                : June 2024
+    copyright            : (C) 2024 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsideoptions.h"
+#include "qgsapplication.h"
+#include "qgssettings.h"
+#include "qgis.h"
+
+#include <QDesktopServices>
+#include <QUrl>
+
+//
+// QgsIdeOptionsWidget
+//
+
+QgsIdeOptionsWidget::QgsIdeOptionsWidget( QWidget *parent )
+  : QgsOptionsPageWidget( parent )
+{
+  setupUi( this );
+
+  layout()->setContentsMargins( 0, 0, 0, 0 );
+
+  connect( mGenerateTokenButton, &QPushButton::clicked, this, &QgsIdeOptionsWidget::generateGitHubToken );
+
+  QgsSettings settings;
+  mTokenLineEdit->setText( settings.value( "pythonConsole/accessTokenGithub", QString() ).toString() );
+}
+
+QgsIdeOptionsWidget::~QgsIdeOptionsWidget() = default;
+
+QString QgsIdeOptionsWidget::helpKey() const
+{
+  return QStringLiteral( "introduction/qgis_configuration.html#code-editor-options" );
+}
+
+void QgsIdeOptionsWidget::apply()
+{
+  QgsSettings settings;
```

Comment:
Am I right that old way to access settings is used for backward compatibility? The new settings key seems not used anywhere.

Reply:
Oops, that was a leftover. Removed now.

I've stuck with the old settings key as most of gui is still using raw QgsSettings access, and I didn't want to have to handle the complexity of forward/backward compatibility.